tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -327,6 +327,39 @@ template <typename HIST = Hist_t>
 class FillParHelper : public RActionImpl<FillParHelper<HIST>> {
    std::vector<HIST *> fObjects;
 
+   void UnsetDirectoryIfPossible(TH1 *h) {
+      h->SetDirectory(nullptr);
+   }
+
+   void UnsetDirectoryIfPossible(...) {}
+
+   // Merge overload for types with Merge(TList*), like TH1s
+   template <typename H>
+   auto Merge(std::vector<H *> &objs, int /*toincreaseoverloadpriority*/)
+      -> decltype(objs[0]->Merge((TList *)nullptr), void())
```

Comment:
Is this done to account for potential cases where `Merge` does not return void? Which would they be?

Reply:
custom types that implement a `Merge` function with this signature but different return value. or in case `Merge` ever changes return type. But in practice the first argument to decltype is just for SFINAE and the second is to clearly document/specify the desired return type